fix: Icon only navigation item padding

balco0110 opened the pull request:
<!-- ## Title: Please consider adding the [skip chromatic] flag to the PR title in case you dont need chromatic testing your changes. -->
## Description:
<!-- *PR notes: Please describe the changes in this PR.* -->
Closes https://github.com/solid-design-system/solid/issues/2250
## Definition of Reviewable:
<!-- *PR notes: Irrelevant elements should be removed.* -->

<!-- *PR notes: If this PR includes a BREAKING CHANGE, a migration guide is needed to explain how users can migrate between versions. * -->
- [ ] E2E tests (features, a11y, bug fixes) are created/updated
<!-- *If this PR includes a bug fix, an E2E test is necessary to verify the change. If the fix is purely visual, ensuring it is captured within our chromatic screenshot tests is sufficient.* -->
- [ ] Stories (features, a11y) are created/updated
- [x] relevant tickets are linked

balco0110 commented on packages/components/src/components/navigation-item/navigation-item.ts:
```diff
@@ -219,9 +219,9 @@ export default class SdNavigationItem extends SolidElement {
+                    <slot part="content" class=${cx('inline-flex items-center', this.vertical && 'gap-2')}></slot>
                   </a>`
-                : html`<slot part="content" class="inline-flex"></slot>`
+                : html`<slot part="content" class=${cx('inline-flex items-center', this.vertical && 'gap-2')}></slot>`
```
i reverted the changes and added new logic

MartaPintoTeixeira reviewed (changes requested):
See comments in cromatic.
Check before and after from the screenshots test please.

balco0110 commented:
> I cant review cromatic. Are the changes done already? Or should it be in progress? Please assign again when cromatic is available.

there were no changes since the last commit
